### DISCUSSION TITLE
How to setup Paisa behind a Reserve Proxy?

### DISCUSSION THREAD

**Discussion by azzamsa** [ananthakumaran/paisa#229] [Q&A] [answered]
Hi, 👋🏽

I have successfully set up [Atuin](https://github.com/atuinsh/atuin) and [TimeTagger](https://github.com/almarklein/timetagger). But I am having a hard time setting up Paisa.

```conf
kit.azzamsa.com {
        respond "Hello, Azzam S.A! 👋🏼"

        handle /timetagger* {
                reverse_proxy localhost:8081
                redir /timetagger /timetagger/
        }

        handle_path /atuin* {
                reverse_proxy localhost:8082
                redir /atuin /atuin/
        }

        handle /paisa* {
                reverse_proxy localhost:8083
                redir /paisa /paisa/
        }
}
```

```conf
services:
  paisa:
    image: docker.io/ananthakumaran/paisa:latest
    ports:
      - "8083:7500"
    volumes:
      - ./data:/root/Documents/paisa/
```

But I am getting `Loading module from “https://kit.azzamsa.com/_app/immutable/chunks/scheduler.rpkcuUjt.js” was blocked because of a disallowed MIME type (“text/plain”).`

![image](https://github.com/ananthakumaran/paisa/assets/17734314/89f28cce-2e2a-4cd4-97f3-b5320fe00483)

Am I missing something?

**Comment by ananthakumaran** [comment]
```
        handle /paisa* {
                reverse_proxy localhost:8083
                redir /paisa /paisa/
        }
```

This part might not work. The initial page will load, but then paisa will assume it's running on a root URL `https://kit.azzamsa.com` and will try to load js/css etc from there. I would suggest setting it up on a subdomain (which proxies everything to paisa), which would be straightforward.

**Reply by azzamsa** [reply]
Do you have any idea how to setup `subpath` instead of `subdomain`?

**Reply by ananthakumaran** [reply, marked as answer]
One way could be to write a rule like `handle /__app*` and route that to paisa backend and so forth. You might have to do the same for static assets and API endpoints. Paisa doesn't natively support hosting on `subpath` as of today.

**Reply by azzamsa** [reply]
Worked like a charm.

```conf
kit.azzamsa.com {
        respond "Hello from kit, Azzam S.A! 👋🏼"

        handle /timetagger* {
                reverse_proxy localhost:8081
                redir /timetagger /timetagger/
        }

        handle_path /atuin* {
                reverse_proxy localhost:8082
                redir /atuin /atuin/
        }
}

paisa.azzamsa.com {
        reverse_proxy localhost:8083
}
```
